Library/Homebrew/formula_installer.rb

```diff
@@ -488,9 +488,9 @@ def expand_dependencies(deps)
 
       if dep.prune_from_option?(build)
         Dependency.prune
-      elsif include_test? && dep.test? && !dep.installed?
-        Dependency.keep_but_prune_recursive_deps
-      elsif dep.build? && install_bottle_for?(dependent, build)
+      elsif dep.test? && !dep.build? && !include_test?
+        Dependency.prune
+      elsif dep.build? && !dep.test? && install_bottle_for?(dependent, build)
```

Comment:
What about if they are both build and test? We don't seem to ever prune in this scenario.